classes/class-settings.php

```diff
@@ -1068,17 +1068,24 @@ public function get_settings_translations( $labels ) {
 	/**
 	 * Update actions dropdown options based on the connector selected.
 	 */
-	function update_actions() {
-		$connector = sanitize_text_field( $_POST['connector'] );
-		$actions = $GLOBALS['wp_stream']->connectors->term_labels['stream_action'];
-		if ( ! empty( $connector ) ) {
-			$class_name = '\WP_Stream\Connector_'.ucfirst( $connector );
-			if ( class_exists( $class_name ) ) {
-				$connector_class = new $class_name();
-				$actions         = $connector_class->get_action_labels();
+	function get_actions() {
+		$connector_name = wp_stream_filter_input( INPUT_POST, 'connector' );
+
+		if ( empty( $connector_name ) ) {
+			wp_send_json_error();
+		}
+
+		$actions = wp_stream_get_instance()->connectors->term_labels['stream_action'];
+
+		$connector_class = '\WP_Stream\Connector_' . ucfirst( $connector_name );
```

Comment:
This won't work for multiname connectors, such as `Connector_User_Switching`, and it won't work if the connector's `name` property doesn't match it's class name.